server/game/cards/02_SHD/leaders/BoKatanKryzePrincessInExile.ts

```diff
@@ -0,0 +1,61 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { Trait, WildcardCardType } from '../../../core/Constants';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { AttacksThisPhaseWatcher } from '../../../stateWatchers/AttacksThisPhaseWatcher';
+
+export default class BoKatanKryzePrincessInExile extends LeaderUnitCard {
+    private attacksThisPhaseWatcher: AttacksThisPhaseWatcher;
+
+    protected override getImplementationId () {
+        return {
+            id: '7424360283',
+            internalName: 'bokatan-kryze#princess-in-exile',
+        };
+    }
+
+    protected override setupStateWatchers (registrar: StateWatcherRegistrar): void {
+        this.attacksThisPhaseWatcher = AbilityHelper.stateWatchers.attacksThisPhase(registrar, this);
+    }
+
+    protected override setupLeaderSideAbilities () {
+        this.addActionAbility({
+            title: 'If you attacked with a Mandalorian unit this phase, deal 1 damage to a unit',
+            cost: [AbilityHelper.costs.exhaustSelf()],
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.Unit,
+                immediateEffect: AbilityHelper.immediateEffects.conditional({
+                    condition: (_) => this.attacksThisPhaseWatcher.getAttackers((attack) => attack.attacker.hasSomeTrait(Trait.Mandalorian)).length > 0,
+                    onTrue: AbilityHelper.immediateEffects.damage({ amount: 1 }),
+                    onFalse: AbilityHelper.immediateEffects.noAction(),
+                })
+            }
+        });
+    }
+
+    protected override setupLeaderUnitSideAbilities () {
+        this.addOnAttackAbility({
+            title: 'You may deal 1 damage to a unit. If you attacked with another Mandalorian unit this phase, you may deal 1 damage to a unit',
```

Comment:
I messaged you in discord about what the behavior is if we target a unit with both damage but it is defeated by the first one. Depending on how that is handled in the engine, we might want to make some changes here.

I think ideally the behavior I would prefer is that if the unit would be defeated by the first damage we don't allow the user to select it for the second damage, that seems like it would be misleading. Instead they could just pass the second damage if they want.

That would require using `sequential()` and `selectCard()` for these, but first let's see what the current behavior is

Reply:
Sorry to keep waffling on this, it's kind of a complicated situation - but I think we do actually have to use `sequential()` and `selectCard()` here. The reason is not so much targeting as the fact that if the unit dies to the first instance of damage, the second instance of damage should not resolve.

This matters if there was, for example, a new triggered ability that said "**When a friendly unit receives damage: draw a card**. If the target unit dies to the first Bo-Katan damage, then the target player should only draw 1 card, not 2. But the way it is currently, the player would draw 2 cards. So we need to use `sequential()` to force the first damage to resolve before the second one does.

Reply:
Hey so I think we had another miscommunication, apologies for making the explanation so confusing. What you did is not quite what I meant 😅 but I think it's okay, I have a plan to make some changes to the way the engine works so that we can get around this problem.

My final answer: can you please revert the change you did here, it is correct each target resolver to have its own optional: true but the ability itself should not. At that point we will be good.